src/thermopro_ble/parser.py

```diff
@@ -63,6 +71,46 @@ def _start_update(self, service_info: BluetoothServiceInfo) -> None:
         if len(data) < 6:
             return
 
+        if name.startswith("TP96"):
+            MAX_BAT = 2880
+            MIN_BAT = 1600  # ??
+            bat_range = MAX_BAT - MIN_BAT
+            (probe_index, internal_temp, battery, ambient_temp) = UNPACK_SPIKE_TEMP(
+                data
+            )
+            internal_temp = internal_temp - 30
+            ambient_temp = ambient_temp - 30
+            battery_percent = (battery - MIN_BAT) / bat_range
+
+            multiprobe_mapping = MULTIPROBE_DEVICES.get(model, {0: ""})
+            probe_name = multiprobe_mapping.get(probe_index, str(probe_index))
+            if probe_name:
+                self.set_title(
+                    f"{name} {probe_name.capitalize()} "
+                    f"{short_address(service_info.address)}"
+                )
+                self.set_device_name(f"{name} {probe_name.upper()}")
```

Comment:
I think the simpler code is better but I'd I'd display it as `probe_index + 1` since only computers count from 0 and humans count from 1 and names are for humans.

Reply:
Agreed 100% about the `probe_index + 1`.  But I'm running into a slight problem and want to propose some options to solve it.

The issue is single probe devices.  In some cases we return before we can read anything useful out of `manufacturer_data`, and then we don't know the probe index.  This actually comes up in the test you wrote that looks at `TP960R_2`.  That adds two entries to `manufacturer_data` so it causes early return before we have a `probe_index`.  In that case, we return early so we get a title/name without the affix, which is inconsistent.

I see three possible solutions to this:

1) We assume and just affix "Probe 1" if the name starts with `"TP96"` before we return
2) We move that return for missing or too much `changed_manufacturer_data` so it is up above with the other early returns, before we `set_title` and `set_device_name`.
3) We only affix a probe index if it's not the first probe, so we'd get `"TP962R (0000)"`, `"TP962R (0000) Probe 2"`

I think I prefer option #2 unless there's some side effect that I'm not aware of with moving that.

Option #3 is the nicest for single probe devices, but I like it less for multi-probe devices.

Reply:
Option 2 has side effects unfortunately (causes `test_tp393_multi_updates` to fail).

I'm going to go with option 1.